### ISSUE TITLE
`Received unexpected backend message of type ParseComplete`: `by-language/csharp-npgsql` starts failing with CrateDB 4.8.4

### ISSUE THREAD

**Issue by amotl** [crate/cratedb-examples#15]
### About
At GH-14, we discovered that the `csharp-npgsql` program would fail its test suite. While it still worked with CrateDB 4.8.3, it starts croaking with CrateDB 4.8.4.

### Exception
The exception can be reproduced by running those commands:
```
docker run -it --rm --publish=4200:4200 --publish=5432:5432 crate:4.8.4
dotnet test --framework=net5.0
```

```c#
[xUnit.net 00:00:00.93]     demo.tests.DemoProgramTest.TestSystemQueryExample [FAIL]
[xUnit.net 00:00:00.95]     demo.tests.DemoProgramTest.TestBasicConversationExample [FAIL]
[xUnit.net 00:00:00.96]     demo.tests.DemoProgramTest.TestAsyncUnnestExample [FAIL]
  Failed demo.tests.DemoProgramTest.TestSystemQueryExample [1 ms]
  Error Message:
   System.AggregateException : One or more errors occurred. (Received unexpected backend message of type ParseComplete) (The following constructor parameters did not have matching fixture data: DatabaseFixture fixture)
---- System.Exception : Received unexpected backend message of type ParseComplete
---- The following constructor parameters did not have matching fixture data: DatabaseFixture fixture
  Stack Trace:

----- Inner Stack Trace #1 (System.Exception) -----
   at Npgsql.NpgsqlDataReader.ProcessMessage(IBackendMessage msg)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlDataReader.NextResult()
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior)
   at Npgsql.PostgresDatabaseInfo.LoadBackendTypes(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at Npgsql.PostgresDatabaseInfo.LoadPostgresInfo(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at Npgsql.PostgresDatabaseInfoFactory.Load(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at Npgsql.NpgsqlDatabaseInfo.Load(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at Npgsql.NpgsqlConnector.LoadDatabaseInfo(Boolean forceReload, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlConnector.Open(NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
   at Npgsql.ConnectorPool.OpenNewConnector(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
   at Npgsql.ConnectorPool.<>c__DisplayClass38_0.<<Rent>g__RentAsync|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Npgsql.NpgsqlConnection.<>c__DisplayClass41_0.<<Open>g__OpenAsync|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Npgsql.NpgsqlConnection.Open()
   at demo.tests.DatabaseFixture..ctor() in /Users/amo/dev/crate/docs/cratedb-examples/by-language/csharp-npgsql/tests/DemoProgramTest.cs:line 22
----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----
```

### Screenshot

![image](https://user-images.githubusercontent.com/453543/214671222-f148aea0-9c4d-44b0-89d7-d8e5febe85db.png)

-- https://github.com/crate/cratedb-examples/actions/runs/4009337440

**Comment by amotl** [comment]
https://github.com/crate/crate-qa/pull/201 and https://github.com/crate/crate/issues/11920 have been resolved, and npgsql has been updated to 6.0.1 on the `crate-qa` repository with https://github.com/crate/crate-qa/pull/206. So, it may also just be a matter of updating the Npgsql package to make it work again.

**Comment by amotl** [comment]
> So, it may also just be a matter of updating the Npgsql package to make it work again.

This is true. CrateDB>=4.8.4 needs Npgsql=>6. fd69b156 should fix it.
